checks.d/consul.py

```diff
@@ -323,3 +361,69 @@ def check(self, instance):
                         status_value,
                         tags=main_tags+node_tags
                     )
+
+        if perform_network_latency_checks:
+            self.check_network_latency(instance, agent_dc, main_tags)
+
+    def _get_coord_datacenters(self, instance):
+        return self.consul_request(instance, '/v1/coordinate/datacenters')
+
+    def _get_coord_nodes(self, instance):
+        return self.consul_request(instance, 'v1/coordinate/nodes')
+
+    def check_network_latency(self, instance, agent_dc, main_tags):
+
+        datacenters = self._get_coord_datacenters(instance)
+        for datacenter in datacenters:
+            name = datacenter['Datacenter']
+            if name != agent_dc:
+                # Not us, skip
+                continue
+            # Inter-datacenter
+            for other in datacenters:
```

Comment:
So, because we're only interested in `agent_dc == name`, I think that after this for loop completes, in line 406, we can probably `break` the outer for loop as there's no reason to waste any more cycles there - the linear search is done. `datacenters` is probably going to be small so there's no big performance gain, that's true.